src/mlpack/tests/lmnn_test.cpp

```diff
@@ -30,25 +30,27 @@ using namespace ens;
  * The target neighbors function should be correct.
  * point.
  */
-TEST_CASE("LMNNTargetNeighborsTest", "[LMNNTest]")
+TEMPLATE_TEST_CASE("LMNNTargetNeighborsTest", "[LMNNTest]", float, double)
 {
+  typedef TestType ElemType;
+
   // Useful but simple dataset with six points and two classes.
-  arma::mat dataset        = "-0.1 -0.1 -0.1  0.1  0.1  0.1;"
-                             " 1.0  0.0 -1.0  1.0  0.0 -1.0 ";
-  arma::Row<size_t> labels = " 0    0    0    1    1    1   ";
+  arma::Mat<ElemType> dataset = "-0.1 -0.1 -0.1  0.1  0.1  0.1;"
+                                " 1.0  0.0 -1.0  1.0  0.0 -1.0 ";
+  arma::Row<size_t> labels    = " 0    0    0    1    1    1   ";
 
-  Constraints<> constraint(dataset, labels, 1);
+  Constraints<arma::Mat<ElemType>, arma::Row<size_t>> constraint(dataset,
+      labels, 1);
 
   // Calculate norm of datapoints.
-  arma::vec norm(dataset.n_cols);
+  arma::Col<ElemType> norm(dataset.n_cols);
   for (size_t i = 0; i < dataset.n_cols; ++i)
   {
     norm(i) = arma::norm(dataset.col(i));
   }
 
   //! Store target neighbors of data points.
-  arma::Mat<size_t> targetNeighbors =
-      arma::Mat<size_t>(1, dataset.n_cols, arma::fill::zeros);
```

Comment:
This was an interesting way to write the code, wondering why.  :smile:

Reply:
Probably brain fry at the end of the day? :smile: